DirectProgramming/DPC++FPGA/Tutorials/Features/loop_fusion/src/loop_fusion.cpp

```diff
@@ -0,0 +1,280 @@
+
+    buffer buffer_array_1(m_array_1);
+    buffer buffer_array_2(m_array_2);
+
+    event e = q.submit([&](handler &h) {
+      accessor accessor_array_1(buffer_array_1, h, write_only, no_init);
+      accessor accessor_array_2(buffer_array_2, h, write_only, no_init);
+
+      h.single_task<DefaultFusionKernel>([=
+      ]() [[intel::kernel_args_restrict]] {  // NO-FORMAT: Attribute
+        for (size_t j = 0; j < kN; j++) {
+          for (size_t i = 0; i < kM; i++) {
+            accessor_array_1[i] = i;
+          }
+          for (size_t i = 0; i < kM; i++) {
+            accessor_array_2[i] = accessor_array_1[i];
+          }
+        }
+      });
+    });
+
+    auto kernel_time = KernelRuntime(e);
+
+    // kernel consists of two loops with one array assignment in each.
+    int num_ops_per_kernel = 2 * kN * kM;
+    std::cout << "Throughput for kernel with default loop fusion and with "
+                 "arrays of size "
+              << kM << ": " << ((double)num_ops_per_kernel / kernel_time)
+              << " Ops/ns\n";
+
+  } catch (sycl::exception const &e) {
+    ErrorReport(e);
+  }
+}
+
+// Does not fuse inner loops because of the [[intel::nofusion]] attribute. Were
+// this attribute not present, the loops would fuse by default.
+void NoFusion(std::array<int, kM> &m_array_1, std::array<int, kM> &m_array_2) {
+  try {
+    queue q(selector, dpc_common::exception_handler,
+            property::queue::enable_profiling{});
+
+    buffer buffer_array_1(m_array_1);
+    buffer buffer_array_2(m_array_2);
+
+    event e = q.submit([&](handler &h) {
+      accessor accessor_array_1(buffer_array_1, h, write_only, no_init);
+      accessor accessor_array_2(buffer_array_2, h, write_only, no_init);
+
+      h.single_task<NoFusionKernel>([=
+      ]() [[intel::kernel_args_restrict]] {  // NO-FORMAT: Attribute
+        for (size_t j = 0; j < kN; j++) {
+          [[intel::nofusion]]  // NO-FORMAT: Attribute
+          for (size_t i = 0; i < kM; i++) {
+            accessor_array_1[i] = i;
+          }
+          for (size_t i = 0; i < kM; i++) {
+            accessor_array_2[i] = accessor_array_1[i];
+          }
+        }
+
+      });
+    });
+
+    auto kernel_time = KernelRuntime(e);
+
+    // kernel consists of two loops with one array assignment in each.
+    int num_ops_per_kernel = 2 * kN * kM;
+    std::cout << "Throughput for kernel with the nofusion attribute and with "
+                 "arrays of size "
+              << kM << ": " << ((double)num_ops_per_kernel / kernel_time)
+              << " Ops/ns\n";
+
+  } catch (sycl::exception const &e) {
+    ErrorReport(e);
+  }
+}
+
+// Does not fuse inner loops by default, since the trip counts are different.
+void DefaultNoFusion(std::array<int, kM + 1> &m_array_1,
+                     std::array<int, kM> &m_array_2) {
+  try {
+    queue q(selector, dpc_common::exception_handler,
+            property::queue::enable_profiling{});
+
+    buffer buffer_array_1(m_array_1);
```

Comment:
Same comment as above. See how you can simplify the code naming when having data, buffer, and accessor.